components/tabs/src/schema.rs

```diff
@@ -38,7 +38,7 @@ const CREATE_PENDING_REMOTE_DELETE_TABLE_SQL: &str = "
         time_sent               INTEGER -- local timestamp, non-null == no longer pending.
     );
 
-    CREATE UNIQUE INDEX IF NOT EXISTS remote_tab_commands_index ON remote_tab_commands(device_id, command, url);
+    CREATE UNIQUE INDEX IF NOT EXISTS remote_tab_commands_index ON remote_tab_commands(device_id, command, IFNULL(url, ''));
```

Comment:
Hmmm, would you be open to modeling this a little differently? It's a more invasive change, but it got me thinking about a more general way to handle commands that aren't tied to URLs, especially since we haven't shipped closing tabs yet:

```sql
CREATE TABLE IF NOT EXISTS remote_tab_commands (
    id                      INTEGER PRIMARY KEY,
    device_id               TEXT NOT NULL,
    command                 INTEGER NOT NULL, -- a CommandKind value
    time_requested          INTEGER NOT NULL, -- local timestamp when this was initially written.
    time_sent               INTEGER -- local timestamp, non-null == no longer pending.
);

CREATE TABLE IF NOT EXISTS remote_tab_command_urls (
    command_id INTEGER NOT NULL REFERENCES remote_tab_commands(id) ON DELETE CASCADE,
    url TEXT NOT NULL,
    PRIMARY KEY (command_id, url)
) WITHOUT ROWID;
```

The `remote_tab_command_urls` table takes the place of our `remote_tab_commands_index` index, and lets us send multiple URL-less commands to the same device.

Reply:
> this design would let us send multiple URL-less commands to the same device.

I don't quite get this part - I don't think we want to queue multiple "CloseAllInactiveTabs" commands any more than we want to queue multiple CloseTab commands for the same URL. I don't see how this world would let us enforce that without significant changes that might seem overkill at face value?

Reply:
I definitely like the idea of potentially having just a queue of any kind of "remote tab management" into the queue. However I wonder if we should move that to a new issue while we can discuss the implications of that! I agree that there's nothing yet that would indicate we'd have more than "CloseAllInactive" (move tabs maybe?? heh). Don't have a strong opinion either way but thought I'd give me 2c!

Reply:
Lina and I chatted about this and agreed to not do this at the current time and instead to wait an see if a future command presents another opportunity to model this differently.